### PR TITLE
Operator SDK bump to 1.2.0

### PR DESCRIPTION
**Pull request by MStokluska**
## Overview

Jira: https://issues.redhat.com/browse/MGDAPI-1130

Due to the number of changes done, here's a list of items that I would like someone to particularly look at:

### Permissions
Rbac permissions have been set in the postgres reconciler - although this does not break the generation or role.yaml file nor it affects how the permissions are working, I believe it would be better to stick the appropriate permissions to appropriate reconcilers.

### Reconciler setup
Reconciler setup has changes, going from the main file to the reconciler, I am not confident enough if each of the reconcilers structs are created correctly, they are working, but maybe there's a better way of doing it.
Cloud metrics reconciler setup, especially around SetupWithManager function. 

### Makefile
Verification around bundles needs a careful review, the Update_CRO_rhmi.md might be followed to check if it makes sense.

### CRO local verification
- run `make cluster/prepare` and `make run` to run CRO locally
- run `make test/e2e/local` in another terminal window and verify that all tests are green

### RHOAM initial install
Because the way operator-sdk deals with prometheus, there are 2 additional files generated in the bundles, a service monitor and a service, without these present, metrics will be scraped but not visible in prometheus. In order to support 0.24.0 version of CRO (bumped version), we need to install CRO via bundle format.
- Delete the "0.23.0" from PREVIOUS_OPERATOR_VERSIONS from the makefile - this should contain all previous versions that the bundle should replace, since we are going with initial install, we won't need these.
- Make sure to point to your own ORG in the makefile and run `make image/build` and once you are happy with the path, push the image to your repo with docker push, alternatively run `make image/build/push`
- In 0.24.0 CSV delte the replaces field and point to your own image of CRO
- Run `make create/olm/bundle` to create and push a bundle and index and verify that the bundle is correct (for example, if the replaces field, if present, replaces an existing bundle)
- Once the index is ready, from master branch of integreatly operator update the logic in: https://github.com/integr8ly/integreatly-operator/blob/master/products/installation.yaml#L64 to:
```
  cloud-resources:
    channel: "rhmi"
    installFrom: "index"
    package: "rhmi-cloud-resources"
    index: "quay.io/_YOUR_INDEX_URL_"
```
- Run RHOAM, ideally on a byoc cluster and confirm that the resources are provisioned and deleted.

### RHOAM upgrade scenario
- Install RHOAM via olm on a byoc cluster
- Change the VERSION field in the makefile to 0.23.0
- Remove the PREVIOUS_OPERATOR_VERSIONS values so that it is empty
- Remove the replaces field from 0.23.0 CSV
- Make sure the ORG in the makefile is matching the desired organization
- Run `make create/olm/bundle` - this will create the initial 0.23.0 bundle and index and push to your repo - we need this in order for 0.24.0 to replace it.
- Update VERSION to 0.24.0 and PREVIOUS_OPERATOR_VERSION to 0.23.0
- Build and push an image with `make image/build/push` - alternatively build first with `make image/build` and push with docker push.
- Udpate 0.24.0 CSV to point to your image and also, confirm that it has the replaces field of 0.23.0
- Go to RHOAM and update the  logic in: https://github.com/integr8ly/integreatly-operator/blob/master/products/installation.yaml#L64 to:
```
  cloud-resources:
    channel: "rhmi"
    installFrom: "index"
    package: "rhmi-cloud-resources"
    index: "quay.io/_YOUR_INDEX_URL_"
```
- Generate new version of RHOAM, for example, 1.5.0
- Build image of rhoam and push it to your quay repo
- Update RHOAM 1.5.0 CSV to point to your image and ensure it replaces current installation of RHOAM
- Follow instructions here to build the bundle and index to simulate RHOAM upgrade scenario: https://github.com/RHCloudServices/integreatly-help/blob/master/guides/olm/installing-rhmi-bundle-format.md
- Run the upgrade and check if the CRO CSV points to a new image, you can also check if the rhmi-registry pod under redhat-rhoam-cloud-resource operator point to a new index
- Verify that the upgrade has been successful
- Delete postgres and redis resource for 3scale and verify that they are deleted and recreated
- Delete rhoam and ensure that the deletion is successful

### CODE REVIEW

**Comment by austincunningham** [comment]
testing locally , s3/blobstorage and RDS/Postgres installing without issue
seeing some error when creating and deleting Redis
e.g.
```
"reconcilerKind": "Redis", "controller": "redis", "name": "example-redis", "namespace": "cloud-resource-operator", "error": "failed to update resource status phase and message: Unauthorized", "errorVerbose": "Unauthorized\nfailed to update resource status phase and message\ngithub.com/integr8ly/cloud-resource-operator/pkg/resources.UpdatePhase\n\t/home/austincunningham/repo-integr8ly/cloud-resource-operator/pkg/resources/phase.go:31\ngithub.com/integr8ly/cloud-resource-operator/controllers/redis.
```

File: Makefile [hunk 1]
```diff
@@ -28,42 +33,39 @@ else
...
-	@$(OPERATOR_SDK) generate csv --operator-name=cloud-resources --csv-version $(VERSION) --from-version $(PREV_VERSION) --make-manifests=false --update-crds --csv-channel=integreatly --default-channel --verbose
-	@sed -i.bak 's/$(PREV_VERSION)/$(VERSION)/g' deploy/olm-catalog/cloud-resources/cloud-resources.package.yaml && rm deploy/olm-catalog/cloud-resources/cloud-resources.package.yaml.bak
-	@sed -i.bak s/cloud-resource-operator:v$(PREV_VERSION)/cloud-resource-operator:v$(VERSION)/g deploy/olm-catalog/cloud-resources/$(VERSION)/cloud-resources.v$(VERSION).clusterserviceversion.yaml && rm deploy/olm-catalog/cloud-resources/$(VERSION)/cloud-resources.v$(VERSION).clusterserviceversion.yaml.bak
+	@$(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests --kustomize-dir=config/manifests --output-dir packagemanifests/ --version ${VERSION} --default-channel --channel integreatly
+	@sed -i "s/Version = \"${PREV_VERSION}\"/Version = \"${VERSION}\"/g" version/version.go
+	@yq w -i "packagemanifests/${VERSION}/cloud-resource-operator.clusterserviceversion.yaml" metadata.annotations.containerImage ${OPERATOR_IMG}
```
Comment (austincunningham) [inline Makefile:63]: @MStokluska can you add yq and version as a dependency in the readme

Comment (MStokluska) [reply Makefile:63]: I've added the `yq` to pre-requisites. 
Thanks Austin!

**Review by austincunningham — APPROVED** [review]
/lgtm
/approve

**Pull request merged**